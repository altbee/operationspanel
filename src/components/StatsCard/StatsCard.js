import React from "react";
import PropTypes from "prop-types";
import { Row, Col, Card, CardBody, CardTitle, CardSubtitle } from "reactstrap";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

export const StatsCard = props => {
  const { icon, className, statsText, statsValue, small } = props;
  const iconSize = small ? "20px" : "45px";
  return (
    <Card className="border-0">
      <CardBody>
        <Row>
          <Col className={small ? "text-center" : ""}>
            <h1
              style={{ fontSize: iconSize }}
              className={small ? "" : "mr-3 mb-0"}
            >
              <FontAwesomeIcon icon={icon} className={className} />
            </h1>
            {!small && (
              <CardTitle className="h6 font-weight-light text-muted">
                {statsText}
              </CardTitle>
            )}
            <CardSubtitle className={small ? "h6" : "h4"}>
              {statsValue}
            </CardSubtitle>
          </Col>
        </Row>
      </CardBody>
    </Card>
  );
};

StatsCard.propTypes = {
  icon: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
  statsText: PropTypes.string.isRequired,
  statsValue: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
    .isRequired,
  small: PropTypes.bool
};

StatsCard.defaultProps = {
  small: false
};

export default StatsCard;
