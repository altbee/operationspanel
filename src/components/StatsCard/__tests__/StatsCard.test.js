import React from "react";
import { configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { library } from "@fortawesome/fontawesome-svg-core";
import { faCircleNotch } from "@fortawesome/free-solid-svg-icons";
import { StatsCard } from "../StatsCard";

library.add(faCircleNotch);

configure({ adapter: new Adapter() });

describe("<StatsCard />", () => {
  const props = {
    statsText: "Hello World",
    icon: "circle-notch",
    statsValue: 5,
    className: "icon-class-name"
  };

  it("accepts props and renders", () => {
    const rendered = mount(<StatsCard {...props} />);
    expect(
      rendered.find(`FontAwesomeIcon[icon='${props.icon}']`).exists()
    ).toBe(true);
    expect(rendered.find(`FontAwesomeIcon.${props.className}`).exists()).toBe(
      true
    );
    expect(rendered.find("CardTitle").text()).toEqual(props.statsText);
    expect(rendered.find("CardSubtitle").text()).toEqual(
      props.statsValue.toString()
    );

    rendered.setProps({ statsValue: props.statsText });
    expect(rendered.find("CardSubtitle").text()).toEqual(props.statsText);
  });

  it("matches the snapshot", () => {
    const tree = mount(<StatsCard {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
