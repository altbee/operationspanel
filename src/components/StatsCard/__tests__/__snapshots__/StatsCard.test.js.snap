// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatsCard /> matches the snapshot 1`] = `
<StatsCard
  className="icon-class-name"
  icon="circle-notch"
  small={false}
  statsText="Hello World"
  statsValue={5}
>
  <Card
    className="border-0"
    tag="div"
  >
    <div
      className="border-0 card"
    >
      <CardBody
        tag="div"
      >
        <div
          className="card-body"
        >
          <Row
            tag="div"
          >
            <div
              className="row"
            >
              <Col
                className=""
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="col"
                >
                  <h1
                    className="mr-3 mb-0"
                    style={
                      Object {
                        "fontSize": "45px",
                      }
                    }
                  >
                    <FontAwesomeIcon
                      border={false}
                      className="icon-class-name"
                      fixedWidth={false}
                      flip={null}
                      icon="circle-notch"
                      inverse={false}
                      listItem={false}
                      mask={null}
                      pull={null}
                      pulse={false}
                      rotation={null}
                      size={null}
                      spin={false}
                      symbol={false}
                      title=""
                      transform={null}
                    >
                      <svg
                        aria-hidden="true"
                        className="svg-inline--fa fa-circle-notch fa-w-16 icon-class-name"
                        data-icon="circle-notch"
                        data-prefix="fas"
                        focusable="false"
                        role="img"
                        style={Object {}}
                        viewBox="0 0 512 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M288 39.056v16.659c0 10.804 7.281 20.159 17.686 23.066C383.204 100.434 440 171.518 440 256c0 101.689-82.295 184-184 184-101.689 0-184-82.295-184-184 0-84.47 56.786-155.564 134.312-177.219C216.719 75.874 224 66.517 224 55.712V39.064c0-15.709-14.834-27.153-30.046-23.234C86.603 43.482 7.394 141.206 8.003 257.332c.72 137.052 111.477 246.956 248.531 246.667C393.255 503.711 504 392.788 504 256c0-115.633-79.14-212.779-186.211-240.236C302.678 11.889 288 23.456 288 39.056z"
                          fill="currentColor"
                          style={Object {}}
                        />
                      </svg>
                    </FontAwesomeIcon>
                  </h1>
                  <CardTitle
                    className="h6 font-weight-light text-muted"
                    tag="h5"
                  >
                    <h5
                      className="h6 font-weight-light text-muted card-title"
                    >
                      Hello World
                    </h5>
                  </CardTitle>
                  <CardSubtitle
                    className="h4"
                    tag="h6"
                  >
                    <h6
                      className="h4 card-subtitle"
                    >
                      5
                    </h6>
                  </CardSubtitle>
                </div>
              </Col>
            </div>
          </Row>
        </div>
      </CardBody>
    </div>
  </Card>
</StatsCard>
`;
